src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/ancm_error_page.htm

```diff
@@ -65,16 +65,15 @@ <h1> HTTP Error %d.%d - %s </h1>
         %s
 
         <h2> Troubleshooting steps: </h2>
-
         <ul>
             <li> Check the system event log for error messages </li>
```

Comment:
I would love if each of these could be deep links into the corresponding help topic.

Reply:
That'd be cool. If we do this, we should make sure the deep link is actually stable and not based on an arbitrary section name or something that can change.

The section name doesn't even need to actually change. I've noticed that our docs generator likes to append a "-1" or "-2" suffix to the actual deep link if the doc has undergone major revisions causing the section to be "repeated" in the source. For example:

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-6.0#configurationloader-1
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-5.0#configurationloader-2

It's super annoying because it makes it impossible to reliably provide a deep link for the latest version of a given doc. @JamesNK @Rick-Anderson @guardrex

Reply:
We have a workaround for the necessary evil of Whole-topic Versioning. Rick likes the anchor tag bookmark, and I like the `id` on an explicit H2 approach ... either way, we can have firm deep bookmarks that are stable. It may only get a little tricky to maintain for future releases with future duplicated content in the topic.

Reply:
Out of curiosity, why add `-1`, etc to titles? It makes linking to titles harder. What problem is it solving?

Reply:
With our "Whole-topic Versioning" approach, markdown headings are duplicated. Bookmarks in DocFX make the `id` from that text. Not knowing what to do with dups, the build system enumerates them down the doc ... and breaks bookmarks accordingly.

The solution is to place versioned content in different files, which I call "Version by File" (MS docs has a different name). That would fix this all permanently everywhere, but it's with the docs team if they wish to pursue it. I tried to implement it, but failed ... I had to revert Blazor docs back to the whole-topic approach.

Reply:
It seems like it would be possible to check if the content is used, e.g. a doc containing duplicated aspnetcore50 and aspnetcore60 sections will only show the one selected. Then only use headings from the used section are processed, they're no longer duplicated, and headings don't need `-1`, etc added to the id.

Although maybe there is something about docfx that makes that not possible. Could you pass along feedback that this is annoying? 😄 

Using one of the workarounds is an ok solution. We'll just need to update aspnetcore.docs with stable links where needed.

Reply:
> Although maybe there is something about docfx that makes that not possible.

Correct, it's baked into their design of the build system. Our use of Whole-topic Versioning ... using the moniker range tags the way we do ... is not how they were designed/intended for use. The approach is very helpful here and saving a lot of 💰 in doc maintenance, but it is buggy WRT the broken link problem ... I would even go as far as saying "embarrassing" ... i.e., Microsoft can't build a working website. 🙈 😄  

It's well known and has been discussed off and on for years. I think Rick and Wade know a lot more about any internal discussions. I'm up for whatever the docs folks can work out, but I won't work directly on the problem again after my 💥  earlier... that's Aquent's official position on it. 🤣 LOL